pageserver/src/tenant.rs

```diff
@@ -3101,7 +3118,23 @@ impl TenantShard {
                 return Ok(GcResult::default());
             }
 
-            if conf.is_gc_blocked_by_lsn_lease_deadline() {
+            // Skip GC if we're within lease deadline.
+            //
+            // Rust & Python tests set single-digit second gc_period and/or
+            // do immediate gc via mgmt API. The lease deadline is hard-coded to
+            // 10min, which would make most if not all tests skip GC here.
+            // Rust tests could in theory tokio::time::advance, but Python
+            // tests have no such options.
+            // Since lease deadline is a crutch we hopefully eventually replace
+            // with durable leases, take a shortcut here and skip lease deadline check
+            // for all tests.
+            // Cf https://databricks.atlassian.net/browse/LKB-92?focusedCommentId=6722329
+            let ignore_lease_deadline = match ignore_lease_deadline {
```

Comment:
Why not just pass an explicit ignore_lead_deadline=true from http.py in the tests, and eliminate the need for this (somewhat subtle) Default behavior?

Reply:
Because of the Rust tests.

Reply:
Sure, but those could just pass ::Yes as well?  Explicit is generally better than implicit.

Reply:
Let’s call it ::InTests instead of ::Default